src/mozanalysis/metrics/desktop.py

```diff
@@ -27,6 +29,32 @@
     experiments_column_type='native',
 )
 
+main = DataSource(
+    name='main',
+    from_expr=dedent("""\
+        (
+            SELECT
+                *,
+                DATE(submission_timestamp) AS submission_date,
```

Comment:
Now that timestamps are more readily available (?) we should consider whether we switch our analysis window logic from dates to timestamps - which would make life simpler in several ways...

Reply:
They'll never be present in clients_daily, though I think e.g. main_summary has them.

Reply:
whispers softly: _we could stop using `clients_daily`_

Reply:
Well, more to the point, they're also in `search_clients_daily`. We should* use UDFs on telemetry.main instead someday, but I'd prefer to kick that can down the road a ways.

* citation needed